### PR TITLE
List books in homepage

### PR DESCRIPTION
**Pull request by NguyenThiHangB**

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+# incrementally modify your database, and then regenerate this schema definition.
+#
+# Note that this schema.rb definition is the authoritative source for your
+# database schema. If you need to create the application database on another
+# system, you should be using db:schema:load, not running all the migrations
+# from scratch. The latter is a flawed and unsustainable approach (the more migrations
+# you'll amass, the slower it'll run and the greater likelihood for issues).
+#
+# It's strongly recommended that you check this file into your version control system.
+
```
Comment (maiq721) [inline db/schema.rb]: delete comment

File: db/seeds.rb [hunk 2]
```diff
@@ -1,7 +1,60 @@
...
-#
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+Category.create!(
+  title: "Ruby",
+  description: "Learning ruby"
+)
+
```
Comment (maiq721) [inline db/seeds.rb]: remove ()

File: db/seeds.rb [hunk 3]
```diff
@@ -1,7 +1,60 @@
...
+  title: "Ruby on rails",
+  description: "The Ruby Programming Language is the authoritative guide to
+   Ruby and provides comprehensive coverage."
+)
+
+Category.create!(
+  title: "PHP",
+  description: "The Ruby Programming Language is the authoritative
+   guide to Ruby and provides comprehensive coverage"
+)
```
Comment (maiq721) [inline db/seeds.rb]: same here

File: db/seeds.rb [hunk 4]
```diff
@@ -1,7 +1,60 @@
...
+
+Book.create!(
+  name: "Rails 5",
+  description: "Rails 5 and Ruby 2.2 bring many improvements, including new APIs and substantial performance enhancements, and the fifth edition of this award-winning classic is now updated!",
+  image: "https://imagery.pragprog.com/products/457/rails5.jpg?1461337189",
+  publish_date: "2016-09-28",
+  price: "200",
+  category_id: 1
+)
```
Comment (maiq721) [inline db/seeds.rb]: same here

File: db/seeds.rb [hunk 5]
```diff
@@ -1,7 +1,60 @@
...
+Book.create!(
+  name: "Ruby on rails",
+  description: "Rails 5 and Ruby 2.2 bring many improvements,
+    including new APIs and substantial performance enhancements,
+    and the fifth edition of this award-winning classic is now updated!",
+  image: "https://imagery.pragprog.com/products/530/rails51.jpg?1492096029",
+  publish_date: "2016-09-28",
+  price: "200",
+  category_id: 2
+)
```
Comment (maiq721) [inline db/seeds.rb]: same here

File: db/seeds.rb [hunk 6]
```diff
@@ -1,7 +1,60 @@
...
+Book.create!(
+  name: "Rails 5",
+  description: "Rails 5 and Ruby 2.2 bring many improvements,
+    including new APIs and substantial performance enhancements,
+    and the fifth edition of this award-winning classic is now updated!",
+  image: "https://imagery.pragprog.com/products/457/rails5.jpg?1461337189",
+  publish_date: "2016-09-28",
+  price: "200",
+  category_id: 1
+)
```
Comment (maiq721) [inline db/seeds.rb]: same here

File: db/seeds.rb [hunk 7]
```diff
@@ -1,7 +1,60 @@
...
+Book.create!(
+  name: "Ruby on rails",
+  description: "Rails 5 and Ruby 2.2 bring many improvements,
+    including new APIs and substantial performance enhancements,
+    and the fifth edition of this award-winning classic is now updated!",
+  image: "https://imagery.pragprog.com/products/530/rails51.jpg?1492096029",
+  publish_date: "2016-09-28",
+  price: "200",
+  category_id: 3
+)
```
Comment (maiq721) [inline db/seeds.rb]: same here

**Comment by NguyenThiHangB** [comment]
ready

**Comment by NguyenThiHangB** [comment]
ready

File: db/seeds.rb [hunk 8]
```diff
@@ -1,7 +1,48 @@
...
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+Category.create! title: "Ruby",
+  description: "Learning ruby"
+
+Category.create! title: "Ruby on rails",
+  description: "The Ruby Programming Language is the authoritative guide to
+   Ruby and provides comprehensive coverage."
```
Comment (mdmahmudurrahman) [inline db/seeds.rb]: You can use faker gem to get arbitary text. 
Reference: [https://github.com/stympy/faker](https://github.com/stympy/faker)

File: db/seeds.rb [hunk 9]
```diff
@@ -1,7 +1,48 @@
...
+    and the fifth edition of this award-winning classic is now updated!",
+  image: "https://imagery.pragprog.com/products/457/rails5.jpg?1461337189",
+  publish_date: "2016-09-28",
+  price: "200",
+  category_id: 1
+
+Book.create! name: "Ruby on rails",
+  description: "Rails 5 and Ruby 2.2 bring many improvements,
+    including new APIs and substantial performance enhancements,
+    and the fifth edition of this award-winning classic is now updated!",
```
Comment (mdmahmudurrahman) [inline db/seeds.rb]: Use faker gem

File: db/seeds.rb [hunk 10]
```diff
@@ -1,7 +1,48 @@
...
+  image: "https://imagery.pragprog.com/products/530/rails51.jpg?1492096029",
+  publish_date: "2016-09-28",
+  price: "200",
+  category_id: 3
+
+
```
Comment (mdmahmudurrahman) [inline db/seeds.rb]: Remove this empty line

**Comment by mdmahmudurrahman** [comment]
commented
Recommended to use faker gem https://github.com/stympy/faker

**Comment by NguyenThiHangB** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 11]
```diff
@@ -173,3 +173,29 @@ footer a:hover {
...
+
+.book {
+  width: 200px;
+  height: 250px;
+}
+
+.content-book {
+  text-align: center;
+  a {
+    color: black;
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Please use color variable.

File: app/assets/stylesheets/custom.scss [hunk 12]
```diff
@@ -173,3 +173,29 @@ footer a:hover {
...
+    &:hover {
+      color: red;
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Same here

File: app/assets/stylesheets/custom.scss [hunk 13]
```diff
@@ -173,3 +173,29 @@ footer a:hover {
...
+      text-decoration: none;
+    }
+  }
+}
+
+.price {
+  font-size: 20px;
+  font-weight: bold;
+  color: red;
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Same here

File: app/views/books/index.html.erb [hunk 14]
```diff
@@ -0,0 +1,12 @@
+<% provide :title, t(".title") %>
+<div class="row">
+  <% @books.each do |book| %>
+    <div class="col-md-3">
+      <%= image_tag book.image, class: "book" %>
+      <div class="content-book">
+        <%= link_to book.name, "#" %>
+        <p class="price"><%= book.price %> <%= t ".$"%></p>
```
Comment (mdmahmudurrahman) [inline app/views/books/index.html.erb]: `<%= t ".$" %>`

File: config/routes.rb [hunk 15]
```diff
@@ -1,8 +1,9 @@
 Rails.application.routes.draw do
+  root "books#index"
   get "/help", to: "static_pages#help"
-  resources :users, only: %i(new create)
   get "/login", to: "sessions#new"
   post "/login", to: "sessions#create"
   delete "/logout", to: "sessions#destroy"
-  root "static_pages#home"
+  resources :users, only: %i(new create)
+  resources :books
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Please specify the action for this resource

File: db/seeds.rb [hunk 16]
```diff
@@ -15,3 +15,19 @@
                password_confirmation: password,
                role: 0
 end
+
+10.times do |n|
+  title = Faker::Lorem.sentence(2)
```
Comment (mdmahmudurrahman) [inline db/seeds.rb]: Please remove () (dap2)

File: db/seeds.rb [hunk 17]
```diff
@@ -15,3 +15,19 @@
...
+  description = Faker::Lorem.sentence(10)
```
Comment (mdmahmudurrahman) [inline db/seeds.rb]: The same here

File: db/seeds.rb [hunk 18]
```diff
@@ -15,3 +15,19 @@
...
+  Category.create! title: title, description: description
+end
+
+categories = Category.take(6)
```
Comment (mdmahmudurrahman) [inline db/seeds.rb]: Same here

File: db/seeds.rb [hunk 19]
```diff
@@ -15,3 +15,19 @@
...
+20.times do
+  name = Faker::Book.title
+  description = Faker::Lorem.sentence(50)
+  image = "https://softcover.s3.amazonaws.com/636/ruby_on_rails_tutorial_4th_edition/images/cover-web.png"
+  publish_date = Faker::Date.forward(23)
```
Comment (mdmahmudurrahman) [inline db/seeds.rb]: The same here

File: db/seeds.rb [hunk 20]
```diff
@@ -15,3 +15,19 @@
...
+  price = Faker::Commerce.price
+  categories.each { |category| category.books.create! name: name, description: description, image: image, publish_date: publish_date, price: price }
```
Comment (mdmahmudurrahman) [inline db/seeds.rb]: Remove the white space after the start and before the end of curly brace

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by NguyenThiHangB** [comment]
ready

File: app/models/book.rb [hunk 21]
```diff
@@ -1,9 +1,12 @@
...
+  belongs_to :categories, optional: true
   has_many :author_books
   has_many :rates
   has_many :comments
   has_many :likes
   has_many :mark_books
   has_many :order_detail
+
+  scope :list_book, -> {select(:id, :name, :image, :price,
+    :created_at, :category_id)}
```
Comment (mdmahmudurrahman) [inline app/models/book.rb]: Please remove optional ()

**Comment by mdmahmudurrahman** [comment]
commented
One more comment to fix.

**Comment by mdmahmudurrahman** [comment]
commented
This (pr) can be (merged)
CC: [To:1818111] Nguyen Thi Huong (94)

**Pull request merged**